ARCs/arc-0018.md

```diff
@@ -0,0 +1,696 @@
+
+If any transfer is initiated by an address that is _not_ listed as the `authorized_address` for this asset ID from this account, the transfer **MUST** be rejected.
+
+If this method is called when there is an existing entry for the same `royalty_asset`, the call is treated as an update.  In the case of an update case the contract **MUST** compare the `current_royalty_asset_amount` and `current_authorized_address` with what is currently set.  If the values differ, the call **MUST** be rejected. 
+
+> This requirement is meant to prevent a sort of race condition where the `authorized_address` has a `transfer` accepted before the `offer`-ing account sees the update. In that case the offering account might try to offer more than they would otherwise want to. An example is offered in [security considerations](#security-considerations)
+
+To rescind an offer, this method is called with 0 as the new offered amount.
+
+If a [transfer](#transfer) or [royalty_free_move](#royalty-free-move) is called successfully, the `offer` **SHOULD** be updated or deleted from local state. 
+
+> Exactly how to update the offer is left to the implementer. In the case of a partially filled offer, the amount may be updated to reflect some new amount that represents `offer_amount - amount transferred` or the offer may be deleted completely.
+
+#### Royalty Free Move (OPTIONAL, ADMIN ONLY):
+
+```
+royalty_free_move(
+    royalty_asset: asset, 
+    royalty_asset_amount: uint64, 
+    from: account,
+    to: account, 
+    current_offered_amount: uint64, 
+    current_auth_address: address
+)
+```
+
+Moves an asset to the new address without collecting any royalty payment. 
+
+Prior to this method being called the current owner **MUST** offer their asset to be moved with the `auth_address` set to the address of the Royalty Enforcer Administrator.
+
+> This May be useful in the case of a marketplace where the NFT must be placed in some escrow account. Any logic may be used to validate this is an authorized transfer.
+
+The `royalty_asset` is the asset being transferred without applying the Royalty Enforcement logic.
+
+The `royalty_asset_amount` is the number of units of this ASA ID that should be moved.
+
+The `from` parameter is the current owner of the asset.
+
+The `to` parameter is the intended receiver of the asset.
+
+The `current_offered_amount` is the number of units of this asset currently offered. This value **MUST** be greater than or equal to the amount being transferred. 
+
+The `current_auth_address` is the authorized address for the current offer. This value **MUST** be the same as the sender of this transaction (the Royalty Enforcer Administrator).
+
+> The `current_*` values for the offer are passed to prevent the race or attack described in [Security Considerations](#security-considerations).
+
+
+### Read Only Methods
+
+Three methods are specified here as `read-only` as defined in ARC22.
+
+#### Get Policy
+```
+get_policy()(address,uint64)
```

Comment:
Should this be named to mirror the setter?

```suggestion
get_royalty_policy()(address,uint64)
```

Alternatively, renamed the setter to (the shorter) `set_policy`.